### PR TITLE
Disable stacktrace unit tests when Trilinos is enabled

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by crtrott** [comment]
This avoids issues with the whole stack trace thing and questions about `-rdynamic` being added.

**Comment by masterleinad** [comment]
What issues are you talking about? Is there anything wrong with `-rdynamic` being added?

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -314,6 +314,7 @@ KOKKOS_ADD_TEST_EXECUTABLE(
 )
 # We need -rdynamic on GNU platforms for the stacktrace functionality
 # to work correctly with shared libraries
+if(NOT KOKKOS_HAS_TRILINOS)
 SET_PROPERTY(TARGET StackTraceTestExec PROPERTY ENABLE_EXPORTS 1)
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: The issue is the target `StackTraceTestExec` is called something else with TriBITS.

Comment (crtrott) [reply core/unit_test/CMakeLists.txt]: We can properly fix it later. For now I just want to get the release out. This seemed like the fastest way of getting Trilinos to pass.

**Pull request merged**